### PR TITLE
Change command system to more extended structure

### PR DESCRIPTION
**Pull request by IceMeltt**

### CODE REVIEW

File: shared/pom.xml [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  <dependencyManagement>
+    <dependencies>
+      <dependency>
+        <groupId>com.whippyteam</groupId>
+        <artifactId>shared-api</artifactId>
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+</project>
```
Comment (kploskonka) [inline shared/pom.xml]: Endline

File: shared/shared-api/pom.xml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    <groupId>pl.tymoteuszboba</groupId>
+    <version>1.0</version>
+  </parent>
+
+  <name>WhippyReloaded || Shared || API</name>
+
+  <artifactId>shared-api</artifactId>
+
+
+</project>
```
Comment (kploskonka) [inline shared/shared-api/pom.xml]: Endline

File: shared/shared-bukkit/pom.xml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+  <dependencies>
+    <dependency>
+      <groupId>pl.tymoteuszboba</groupId>
+      <artifactId>shared-api</artifactId>
+      <version>1.0</version>
+    </dependency>
+  </dependencies>
+
+</project>
```
Comment (kploskonka) [inline shared/shared-bukkit/pom.xml]: Endline

File: shared/shared-api/src/main/java/com/whippyteam/shared/command/CommandContent.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+    String getParams(int from, int to);
+
+    default int getParamsLength() {
+        return this.getArgs().length;
+    }
+
+    String getLabel();
+
+    Command getCommand();
+}
```
Comment (kploskonka) [inline shared/shared-api/src/main/java/com/whippyteam/shared/command/CommandContent.java]: Isn't it better to fit this in Command class? I feel that dividing it is unnecessary.

File: shared/shared-bukkit/src/main/java/com/whippyteam/shared/bukkit/command/BukkitCommandHandler.java [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+            } catch (Throwable throwable) {
+                throwable.printStackTrace();
+            }
+
+        } catch (CommandUnpermittedSenderException exception) {
+            String message = ChatColor.RED + "This command can be executed only by a Player!";
+            sender.sendMessage(message);
+        } catch (CommandPermissionException exception) {
+            String message =
+                ChatColor.RED + "You don't have enough permission to use this command!";
```
Comment (kploskonka) [inline shared/shared-bukkit/src/main/java/com/whippyteam/shared/bukkit/command/BukkitCommandHandler.java]: "You dont' have a permission to use this command!" is better.

File: tools/pom.xml [hunk 6]
```diff
@@ -42,6 +44,11 @@
...
       <artifactId>slf4j-simple</artifactId>
     </dependency>
+
+    <dependency>
+      <groupId>pl.tymoteuszboba</groupId>
+      <artifactId>shared-bukkit</artifactId>
+    </dependency>
   </dependencies>
 
 </project>
```
Comment (kploskonka) [inline tools/pom.xml]: Endline

File: shared/shared-api/src/main/java/com/whippyteam/shared/command/CommandContent.java (hunk 4 above)
Comment (IceMeltt) [reply shared/shared-api/src/main/java/com/whippyteam/shared/command/CommandContent.java]: Great idea!

**Pull request merged**